port/unix/omrsysinfo.c

```diff
@@ -4470,65 +4451,72 @@ omrsysinfo_cgroup_subsystem_iterator_next(struct OMRPortLibrary *portLibrary, st
 {
 	int32_t rc = OMRPORT_ERROR_SYSINFO_CGROUP_SUBSYSTEM_UNAVAILABLE;
 #if defined(LINUX) && !defined(OMRZTPF)
-	BOOLEAN isSingleLineKeyValue = FALSE;
 	struct OMRCgroupMetricInfoElement *currentElement = NULL;
 	const struct OMRCgroupSubsystemMetricMap *subsystemMetricMap = NULL;
 	const struct OMRCgroupSubsystemMetricMap *subsystemMetricMapElement = NULL;
 	if (state->count >= state->numElements) {
 		goto _end;
 	}
 	switch (state->subsystemid) {
-	case OMR_CGROUP_SUBSYSTEM_MEMORY :
+	case OMR_CGROUP_SUBSYSTEM_MEMORY:
 		subsystemMetricMap = omrCgroupMemoryMetricMap;
 		break;
-	case OMR_CGROUP_SUBSYSTEM_CPU :
+	case OMR_CGROUP_SUBSYSTEM_CPU:
 		subsystemMetricMap = omrCgroupCpuMetricMap;
 		break;
-	case OMR_CGROUP_SUBSYSTEM_CPUSET :
+	case OMR_CGROUP_SUBSYSTEM_CPUSET:
 		subsystemMetricMap = omrCgroupCpusetMetricMap;
 		break;
 	default:
 		rc = OMRPORT_ERROR_SYSINFO_CGROUP_SUBSYSTEM_UNAVAILABLE;
+		state->count += 1;
 		goto _end;
 	}
 	subsystemMetricMapElement = &subsystemMetricMap[state->count];
 	currentElement = subsystemMetricMapElement->metricInfoElementList;
 	if (NULL == state->fileContent) {
-		if ((SINGLE_CGROUP_METRIC == subsystemMetricMapElement->metricElementsCount) && (NULL != currentElement->metricKeyInFile)) {
-			isSingleLineKeyValue = TRUE;
-		}
-		rc = readCgroupMetricFromFile(portLibrary, state->subsystemid, subsystemMetricMapElement->metricFileName, currentElement->metricKeyInFile, metricElement->value, &(*state).fileContent, isSingleLineKeyValue, sizeRef);
-		if (0 == rc) {
-			if(NULL == state->fileContent) {
-				state->multiLineCounter = SINGLE_CGROUP_METRIC;
-			} else {
-				state->multiLineCounter = 0;
-			}
-		} else {
+		rc = readCgroupMetricFromFile(portLibrary, state->subsystemid, subsystemMetricMapElement->metricFileName, currentElement->metricKeyInFile, &(state->fileContent), metricElement->value, sizeRef);
+
+		// In error condition or single metric condition we increment the counter to continue to next metric
```

Comment:
`//` comments are not c89 compliant. Can you change these to `/* */` comments?

Reply:
sure @charliegracie will be changing it.